src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -400,7 +417,8 @@ public void setupBuildPlansForNewExercise(ProgrammingExercise programmingExercis
             String script = buildScriptGenerationService.get().getScript(programmingExercise);
             programmingExercise.setBuildPlanConfiguration(new Gson().toJson(windfile));
             programmingExercise.setBuildScript(script);
-            programmingExercise = programmingExerciseRepository.save(programmingExercise);
+            programmingExerciseRepository.saveAndFlush(programmingExercise);
```

Comment:
this is duplicated code that we save and flush and then again load it.
Can you extract a method save(exercise) in this service?

Reply:
And please add some comments why it is necessary to load the exercise again